src/toil/batchSystems/mesos/executor.py

```diff
@@ -98,7 +101,13 @@ def launchTask(self, driver, task):
         def runTask():
             log.debug("Running task %s", task.task_id.value)
             sendUpdate(mesos_pb2.TASK_RUNNING)
-            popen = runJob(pickle.loads(task.data))
+            # This is where task.data is first invoked. Using this position to setup cleanupInfo
+            taskData = pickle.loads(task.data)
+            if hasattr(self, 'workerCleanupInfo'):
```

Comment:
Too complicated. Why not just `self.workerCleanupInfo is not None`? All object attributes should be initialized in the constructor. It's bad style to introduce new attributes after construction. I can explain why face to face. If you don't have a value for an attribute yet, use None. PyCharm should warn you about this, actually.